### PR TITLE
Security Enhancements preventing unauthenticated access

### PR DESCRIPTION
**Pull request by amanpro30**
## Description
Added some additional security headers to APIRequestHandler and protected all the backend API routes from unauthenticated access.

## Reviewers
@7a @viyatb @sharmamohit123 @saganshul @Ashrith-Shetty 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other

### Checklist:

- [ ] My code follows the code style (modified PEP8) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: owtf/api/handlers/base.py [hunk 1]
```diff
@@ -45,6 +45,14 @@ class BaseRequestHandler(RequestHandler):
     def set_default_headers(self):
         self.add_header("X-OWTF-Version", __version__)
         self.add_header("X-Frame-Options", "SAMEORIGIN")
+        self.add_header("X-XSS-Protection", "1; mode=block")
+        self.add_header("X-Content-Type-Options", "nosniff")
+        self.add_header("Strict-Transport-Security", "max-age=31536000; includeSubDomains")
```
Comment (viyatb) [inline owtf/api/handlers/base.py]: I think this will break functionality with https?

Comment (amanpro30) [reply owtf/api/handlers/base.py]: Removed.

File: owtf/webapp/src/containers/WorklistPage/api.js [hunk 2]
```diff
@@ -1,20 +1,22 @@
 import Request from "../../utils/request";
 import { API_BASE_URL } from "../../utils/constants";
 
+const options = {
+  headers: {
+    "Content-Type": "application/x-www-form-urlencoded;charset=UTF-8",
+    Authorization: "Bearer " + localStorage.getItem("token")
```
Comment (viyatb) [inline owtf/webapp/src/containers/WorklistPage/api.js:7]: doesn't this have to be in quotes? `"Authorization"`

**Review by viyatb — CHANGES_REQUESTED** [review]

File: owtf/webapp/src/containers/WorklistPage/api.js (hunk 2 above)
Comment (amanpro30) [reply owtf/webapp/src/containers/WorklistPage/api.js:7]: This is correct and is working fine when tested. The one with quotes is not working and also when I use auto-formatting it is removing the quotes.

Comment (amanpro30) [reply owtf/webapp/src/containers/WorklistPage/api.js:7]: Hi @viyatb,
I searched this on google and found this link: https://stackoverflow.com/questions/60790742/vs-code-node-js-prettier-auto-formatting-messing-up-the-dictionary-key
JS object properties **do not need quotes** except when the property name needs to contain normally invalid characters, like spaces, or dashes, etc.

Comment (viyatb) [reply owtf/webapp/src/containers/WorklistPage/api.js:7]: weird 👍

**Pull request merged**